### PR TITLE
iio: Replace the marco with an inline function

### PR DESCRIPTION
**Pull request by yongli3**
Signed-off-by: Yong Li <yong.b.li@intel.com>

### CODE REVIEW

**Comment by yongli3** [comment]
@edersondisouza @laykuanloon please help to review this PR, it is based on PR #2231 and PR @2215

**Comment by laykuanloon** [comment]
@yongli3 

I have problem when patch this. The sensor scale become 0 causing all my data print out become 0.

WRN: ./src/lib/io/sol-iio.c:559 iio_set_channel_scale() Could not set scale to 0.000000 on channel [in_accel_x] of device0
DBG: ./src/lib/io/sol-iio.c:978 sol_iio_add_channel() channel [in_accel_x] added. scale: 0.000000 - offset: 0 - storagebits: 16 - bits: 12 - mask: 4095

If use PR #2231 then the scale is correct
DBG: ./src/lib/io/sol-iio.c:978 sol_iio_add_channel() channel [in_accel_x] added. scale: 0.019163 - offset: 0 - storagebits: 16 - bits: 12 - mask: 4095

File: src/modules/flow/iio/nodes.c [hunk 1]
```diff
@@ -79,6 +70,20 @@ struct iio_node_type {
...
+static inline struct sol_iio_channel *
+iio_add_channel(double scale, double offset, const char *name, struct iio_device_config *base)
+{
+    struct sol_iio_channel_config channel_config = SOL_IIO_CHANNEL_CONFIG_INIT;
+
+    if (base->use_device_default_scale)
+        channel_config.scale = scale;
+
+    if (base->use_device_default_offset)
+        channel_config.scale = offset;
```
Comment (edersondisouza) [inline src/modules/flow/iio/nodes.c]: Here you probably meant `channel_config.offset = offset`. I guess that explain @laykuanloon scale issue.

**Comment by edersondisouza** [comment]
I'm inclined to go with this patch instead of #2231. Once @yongli3 fixes it it should be fine. @laykuanloon any other comments about it?

**Comment by yongli3** [comment]
Many thanks to @edersondisouza @laykuanloon! I fixed the typo and please help to review it.

**Comment by bdilly** [comment]
Thanks @yongli3 . It seems right
Merged

**Pull request closed without merging**